pkg/secrets/info_nix.go

```diff
@@ -10,37 +10,43 @@ package secrets
 
 import (
 	"fmt"
+	"io"
 	"os/user"
 	"strconv"
 	"syscall"
 )
 
-func (info *SecretInfo) populateRights() {
-	err := checkRights(info.ExecutablePath, secretBackendCommandAllowGroupExec)
+func printExecutateRights(w io.Writer) {
```

Comment:
```suggestion
func printSecretsBackendPermissions(w io.Writer) {
```

Reply:
The pattern of passing in writer this low-level in a helper function vs using an object return is pretty unconventional for Golang and reduces this code's portability.